File: Python/functions/no checking/magic.py
#Magic No.
def magic(n):
    s=0
    while(n>0):
        s+=n%10
        n//=10
        if n==0:
            if s==1:
                return True
            elif s<10:
                return False
            else:
                n=s
                s=0 
n=int(input("Enter the no. ="))
if magic(n):
    print("Magic no.")
else:
    print ("Not Magic No.")

File: JAVA/Number check/Magic.java
import java.util.Scanner;
class Magic
{
    public static int Jog(int a)
    {
      int r=0,l,s=0;
      for(;a!=0;a=a/10)
      {
        l=a%10;
        s+=l;
        r=r*10+l;
      }
      return s;
    }
    public static int Ulto(int j)
    {
      int r=0;
      for(;j!=0;j=j/10)
      {
        r=(r*10)+(j%10);
      }
      return r;
    }
     public static int goon(int k)
    {
        int  i,s;
        i=Jog(k);
        s=Jog(k) * Ulto(i);
        return s;
    }
    public static void main(String as[])
    {
        int m;
        Scanner ob=new Scanner(System.in);
        System.out.print("Enter the no.=");
        m=ob.nextInt();
        System.out.print( (int)m==(int)goon(m) ? "Magic":"Not");
    }
}